src/free.rs

```diff
@@ -105,22 +105,26 @@ pub fn rev<I>(iterable: I) -> iter::Rev<I::IntoIter>
     iterable.into_iter().rev()
 }
 
-/// Iterate `i` and `j` in lock step.
+/// Returns an iterator pairing two iterators together in lock step.
+///
+/// `zip()` returns an iterator where each item consists of items from the _first_ and _second_ iterator. 
+/// The returned iterator has as many items as the shortest of the two iterators passed in.
 ///
 /// [`IntoIterator`] enabled version of [`Iterator::zip`].
+/// 
+/// ## Example
 ///
 /// ```
 /// use itertools::zip;
 ///
-/// let data_1 = [1, 2, 3, 4, 5];
-/// let data_2 = ['a', 'b', 'c'];
 /// let mut result: Vec<(i32, char)> = Vec::new();
 ///
-/// for (a, b) in zip(&data_1, &data_2) {
+/// for (a, b) in zip(&[1, 2, 3, 4, 5], &['a', 'b', 'c']) {
 ///     result.push((*a, *b));
 /// }
 /// assert_eq!(result, vec![(1, 'a'),(2, 'b'),(3, 'c')]);
 /// ```
+#[deprecated(note="Use std::iter::zip instead", since="1.59.0")]
```

Comment:
I think the `since` field refers to the version of `itertools`, not of `std`. (At least that's my impression when reading other `deprecated` notices in `itertools`.)

Please change to `0.10.4`.

Reply:
Makes perfect sense.